source/extensions/certificate_providers/local_certificate/local_certificate.cc

```diff
@@ -211,6 +213,11 @@ void Provider::signCertificate(const std::string sni,
   // Update certificates_ map
   {
     absl::WriterMutexLock writer_lock{&certificates_lock_};
+    // If cache is full, remove first element in the map
+    if (certificates_.size() >= max_hosts_) {
+      certificates_.erase(certificates_.begin());
```

Comment:
This logic is to remove the first generated cert, while it might be used very frequently. Can we design a LRU cache? then we remove the least recently used cert.